packages/assets-controllers/src/NftController.ts

```diff
@@ -52,6 +56,77 @@ type SuggestedNftMeta = {
   origin: string;
 };
 
+export enum OpenSeaV2ChainIds {
+  ethereum = 'ethereum',
+}
+
+export type OpenSeaV2GetNftResponse = { nft: OpenSeaV2DetailedNft };
+
+export type OpenSeaV2Nft = {
+  identifier: string;
+  collection: string;
+  contract: string;
+  token_standard: string;
+  name: string;
+  description: string;
+  image_url: string;
+  metadata_url: string;
+  updated_at: string;
+  is_disabled: boolean;
+  is_nsfw: boolean;
+};
+
+export type OpenSeaV2DetailedNft = OpenSeaV2Nft & {
+  animation_url: string;
+  is_suspicious: boolean;
+  creator: string;
+  traits: {
+    trait_type: string;
+    display_type: string;
+    max_value: string;
+    trait_count: number;
+    value: number | string;
+  }[];
+  owners: {
+    address: string;
+    quantity: number;
+  }[];
+  rarity: { rank: number };
+};
+
+export type OpenSeaV2ListNftsResponse = {
+  nfts: OpenSeaV2Nft[];
+  next?: string;
+};
+
+export type OpenSeaV2Contract = {
+  address: string;
+  chain: string;
+  collection: string;
+  contract_standard: string;
+  name: string;
+  supply: number;
+};
+
+export type OpenSeaV2Collection = {
+  collection: string;
+  name: string;
+  description: string;
+  image_url: string;
+  owner: string;
+  category: string;
+  is_disabled: boolean;
+  is_nsfw: boolean;
+  trait_offers_enabled: boolean;
+  opensea_url: string;
+  project_url: string;
+  wiki_url: string;
+  discord_url: string;
+  telegram_url: string;
+  twitter_username: string;
+  instagram_username: string;
```

Comment:
Some of these are documented as optional in their docs, perhaos we should consider them optional as well?

Unsure if they're omitted in practice


```suggestion
  description?: string;
  image_url?: string;
  owner: string;
  category: string;
  is_disabled: boolean;
  is_nsfw: boolean;
  trait_offers_enabled: boolean;
  opensea_url: string;
  project_url?: string;
  wiki_url?: string;
  discord_url?: string;
  telegram_url?: string;
  twitter_username?: string;
  instagram_username?: string;
```

Reply:
Updated the type optionality according to the openapi spec.

I have seen one discrepancy that the `next` cursor can be omitted despite being in `required: []` so that one is kept optional.